test/tests/functional/pbs_pbsnodes_S.py

```diff
@@ -0,0 +1,99 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2018 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class TestPbsnodesS(TestFunctional):
+
+    """
+    This test suite contains regression tests for pbsnodes -S.
+    """
+    du = DshUtils()
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+
+        self.header = ['vnode', 'state', 'OS', 'hardware', 'host',
+                       'queue', 'mem', 'ncpus', 'nmics', 'ngpus', 'comment']
+        self.pbs_exec = self.server.pbs_conf['PBS_EXEC']
+        self.svrname = self.server.pbs_server_name
+        self.momA = self.moms.values()[0]
+        self.hostA = self.momA.shortname
+
+    def test_pbsnodes_S(self):
+        """
+        This verifies that 'pbsnodes -S' results in a usage message
+        """
+        pbsnodes_S = [os.path.join(self.pbs_exec, 'bin', 'pbsnodes'), '-S']
+        out = self.du.run_cmd(self.svrname, cmd=pbsnodes_S)
+        found = 0
+        if 'usage' in out['err'][0]:
+            found = 1
+        self.assertEqual(found, 1, "usage not found in error message")
+
+    def test_pbsnodes_S_host(self):
+        """
+        This verifies that 'pbsnodes -S <host>' results in an output
+        with correct headers.
+        """
+        pbsnodes_S_host = [
+            os.path.join(self.pbs_exec, 'bin', 'pbsnodes'), '-S', self.hostA]
+        out1 = self.du.run_cmd(self.svrname, cmd=pbsnodes_S_host)
```

Comment:
out1 and out2 are in different tests, no real need to differentiate it.  It's up to you though.

Reply:
print the output to console for debugging purposes